File: cpp/0088合并两个有序数组.cpp
#include<iostream>
#include<vector>
#include<algorithm>
using namespace std;

void merge1(vector<int> &nums1, int m, vector<int> &nums2, int n)
{
    for (int i = 0;i<n;++i)
        nums1[m + i] = nums2[i];

    sort(nums1.begin(), nums1.end());
}

void merge2(vector<int> &nums1, int m, vector<int> &nums2, int n)
{
    vector<int> temp(m);
    for (int i= 0;i<m;++i)
    {
        temp[i] = nums1[i];
    }
    int i = 0, j = 0, k = 0;

    while (i < m and j < n)
    {
        if (temp[i] < nums2[j])
            nums1[k] = temp[i++];
        
        else
            nums1[k] = nums2[j++];

        ++ k;
    }

    if(i==m)
    {
        for (int z = 0;z<n-j;++z)
            nums1[k + z] = nums2[j + z];
    }

        
    if(j==n)
    {
        for (int z = 0; z < m - i; ++z)
            nums1[k + z] = temp[i + z];
    }
}

File: python/0088合并两个有序数组.py
nums=[1,2,3,0,0,0]
m = 3 
nums2 = [2,5,6] 
n = 3

# 合并排序
def merge1(nums1: list, m: int, nums2: list, n: int):
    nums1[m:]=nums2
    nums1.sort()



# 双指针
def merge2(nums1: list, m: int, nums2: list, n: int):
    nums_temp=nums1[:m]
    i=j=k=0

    while(i<m and j<n):
        if(nums_temp[i]<nums2[j]):
            nums1[k]=nums_temp[i]
            i+=1
        else:
            nums1[k]=nums2[j]
            j+=1
        k+=1
    
    if(i==m):
        nums1[k:]=nums2[j:]
    if(j==n):
        nums1[k:]=nums_temp[i:]
    

merge2(nums,m,nums2,n)
print(nums)

